extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -448,17 +448,31 @@ export class SubscriptionService {
    * @return {!Promise<boolean>}
    */
   delegateActionToService(action, serviceId) {
-    // TODO: add a promise to wait for resolve promise of a platform.
-    const platform = dev().assert(this.platformStore_.getPlatform(serviceId),
-        'Platform is not registered');
-    this.subscriptionAnalytics_.event(
-        SubscriptionAnalyticsEvents.ACTION_DELEGATED,
-        {
-          action,
-          serviceId,
-        }
-    );
-    return platform.executeAction(action);
+    return this.platformStore_.whenPlatformResolves(serviceId)
+        .then(platform => {
+          dev().assert(platform, 'Platform is not registered');
+          this.subscriptionAnalytics_.event(
+              SubscriptionAnalyticsEvents.ACTION_DELEGATED,
+              {
+                action,
+                serviceId,
+              }
+          );
+          return platform.executeAction(action);
+        });
+  }
+
+  /**
+   * Delegate UI decoration to another service.
+   * @param {Element} element
+   * @param {string} serviceId
+   */
+  delegateDecorationToElement(element, serviceId) {
```

Comment:
So, this is actually slightly more complicated. The actual actions we take here are: 

1. _Always_ attach an action from another service (attach delegated, or some such).
2. _Optionally_ decorate.

As such, the actual API should read something like this:

```
attachDelegatedActionToElement(element, serviceId, action, opt_decorate)
```

Reply:
Actually, I'm taking some of this back. I think it's fine to call this method something like `decorateServiceAction`. But I figured out what I was missing: where is the new form of `delegateActionToService` called for the specified `subscriptions-service` attribute?